tests/testsuite/support/matchers.rs

```diff
@@ -0,0 +1,734 @@
+use std::fmt;
```

Comment:
Should we pull this out into a separate crate?

Reply:
Yeah, this should probably be a separate crate - I'll pull this stuff out

Reply:
just extracted this to `test_support` crate